src/gleam/os.gleam

```diff
@@ -0,0 +1,45 @@
+//// Function to interact with the host operating system.
+
+import gleam/list
+import gleam/string
+
+// Internal type for erlang interop.
+external type CharList
+
+external fn os_getenv() -> List(CharList) =
+  "os" "getenv"
+
+external fn os_putenv(key: CharList, value: CharList) -> Bool =
+  "os" "putenv"
+
+external fn os_unsetenv(key: CharList) -> Bool =
+  "os" "unsetenv"
+
+external fn char_list_to_string(CharList) -> String =
+  "erlang" "list_to_binary"
+
+external fn string_to_char_list(String) -> CharList =
+  "erlang" "binary_to_list"
+
+/// Return all environment variables set on the system.
+pub fn get_env() -> List(tuple(String, String)) {
```

Comment:
Should this return a map?

Reply:
I don't think you can have duplicate values but I think that ENV variables are ordered.

Reply:
I checked with Erlang, Node, Lua, Elixir, and Ruby, and all returned them in different orders, so I'm don't think they are.

Elixir and JavaScript returned them in un-ordered data structures too